### PR TITLE
M3-4188: Hiding SSH key remove button for LISH settings

### PR DESCRIPTION
**Pull request by tiffwong**
## Description
Hide the "remove" button for SSH Public Key under Profile > LISH if there are no SSH keys

## Type of Change
- Non breaking change ('update', 'change')

### CODE REVIEW

File: packages/manager/src/features/Profile/LishSettings/LishSettings.tsx [hunk 1]
```diff
@@ -211,12 +213,14 @@ class LishSettings extends React.Component<CombinedProps, State> {
                     className={classes.keyTextarea}
                     data-qa-public-key
                   />
-                  <Button
-                    buttonType="remove"
-                    onClick={this.onPublicKeyRemove(idx)}
-                    className={classes.remove}
-                    data-qa-remove
-                  />
+                  {authorizedKeys[idx] && (
```
Comment (Jskobos) [inline packages/manager/src/features/Profile/LishSettings/LishSettings.tsx:216]: If we want to be super pedantic this should be `typeof authorizedKeys[idx] !== 'undefined'`. If you clear the input field manually the "remove" button disappears which is a little weird.

File: packages/manager/src/features/Profile/LishSettings/LishSettings.tsx [hunk 2]
```diff
@@ -289,6 +296,12 @@ class LishSettings extends React.Component<CombinedProps, State> {
     e: React.ChangeEvent<HTMLInputElement>
   ) => {
     this.setState(set(lensPath(['authorizedKeys', idx]), e.target.value));
+
+    if (e.target.value == '') {
```
Comment (Jskobos) [inline packages/manager/src/features/Profile/LishSettings/LishSettings.tsx:300]: This is ok, though it goes against the principle that you shouldn't store in state things that can be derived from other state (or props). You can instead put some logic in the render function to determine if there are no valid keys, something like:

```
const isSSHKeyEmpty = authorizedKeysCount === 1 && authorizedKeys[0] === '';
```

File: packages/manager/src/features/Profile/LishSettings/LishSettings.tsx [hunk 3]
```diff
@@ -211,12 +211,14 @@ class LishSettings extends React.Component<CombinedProps, State> {
                     className={classes.keyTextarea}
                     data-qa-public-key
                   />
-                  <Button
-                    buttonType="remove"
-                    onClick={this.onPublicKeyRemove(idx)}
-                    className={classes.remove}
-                    data-qa-remove
-                  />
+                  {typeof authorizedKeys[0] !== 'undefined' && (
```
Comment (johnwcallahan) [inline packages/manager/src/features/Profile/LishSettings/LishSettings.tsx:214]: One small adjustment: I think this should be `(idx !== 0 || authorizedKeys[0] !== 'undefined)`. Currently, if you click "Add SSH Public Key" without typing in the text area, you don't see any "Remove" buttons unless the first TextArea has text in it.

File: packages/manager/src/features/Profile/LishSettings/LishSettings.tsx [hunk 4]
```diff
@@ -211,12 +211,14 @@ class LishSettings extends React.Component<CombinedProps, State> {
                     className={classes.keyTextarea}
                     data-qa-public-key
                   />
-                  <Button
-                    buttonType="remove"
-                    onClick={this.onPublicKeyRemove(idx)}
-                    className={classes.remove}
-                    data-qa-remove
-                  />
+                  {idx !== 0 && authorizedKeys[0] !== 'undefined' && (
```
Comment (johnwcallahan) [inline packages/manager/src/features/Profile/LishSettings/LishSettings.tsx:214]: If you want the Remove button to appear on the first key, the first `&&` needs to be `||`.

Comment (tiffwong) [reply packages/manager/src/features/Profile/LishSettings/LishSettings.tsx:214]: I thought we didn't want it to appear on the first key

Comment (tiffwong) [reply packages/manager/src/features/Profile/LishSettings/LishSettings.tsx:214]: Also having `||` means that the remove button still shows up even if the key is undefined

**Pull request merged**